Emotion

jonenst opened the pull request.

AbdelHedhili commented on demo/src/right-resizable-box.js:
```diff
@@ -66,27 +68,26 @@ const RightResizableBox = (props) => {
     return (
-        <ResizableBox
+        <ResizableBoxSx
             style={{ display: props.hide ? 'none' : undefined }}
```
why isn't this in the sx ?

jonenst replied:
I migrated makeStyle/useStyle to emotion but left everything else as is. Can be refactored in a separate PR

AbdelHedhili commented on demo/src/app.js:
```diff
@@ -623,6 +657,28 @@ const AppContent = ({ language, onLanguageClick }) => {
                             </div>
                             <div style={{ flexGrow: 1 }} />
                             <div style={{ alignSelf: 'center' }}>baz</div>
```
sx ? Or you don't want to change the div to a box just for that ?

jonenst replied:
I migrated makeStyle/useStyle to emotion but left everything else as is. Can be refactored in a separate PR

AbdelHedhili commented on src/components/MuiVirtualizedTable/MuiVirtualizedTable.js:
```diff
@@ -814,6 +852,7 @@ class MuiVirtualizedTable extends React.PureComponent {
                     height: '100%',
                 }}
+                className={this.props.className}
             >
```
why aren't all the static styles 'sxed' in this part

jonenst replied:
I migrated makeStyle/useStyle to emotion but left everything else as is. Can be refactored in a separate PR

AbdelHedhili reviewed (commented):
code: done
test: in progress

AbdelHedhili reviewed (dismissed):
the topbar style is broken
![toolbar](https://github.com/gridsuite/commons-ui/assets/46495975/37c6e09a-c5d9-481a-b84f-28d6effaf728)

jonenst commented:
fixed margin, with sx it's not pixels when providing a number, it's multiple of theme.spacing (default 8px) https://mui.com/system/getting-started/the-sx-prop/#spacing


fixed report viewer sx was overwritten by parent instead of passing the classname

Walid-Sahnoun reviewed (commented):
logs in left side is broken
there is no horizontal scroll, then the log is written in more than 1 line.
also, the highlight is not showed when you hover on second or third level. 
![image](https://github.com/gridsuite/commons-ui/assets/107688384/d8a32003-5b15-4d8e-b67c-cef1cb447361)

flomillot commented on src/utils/styles.js:
```diff
@@ -0,0 +1,29 @@
+export const toNestedGlobalSelectors = (styles, generateGlobalClass) =>
+    Object.fromEntries(
+        Object.entries(styles).map(([k, v]) => [
+            `& .${generateGlobalClass(k)}`,
```
Why do we have to use this ?

flomillot replied:
I mean, we can't get rid of classNames ?

flomillot replied:
Yes, but almost everytime, we stylise the component itself and not nested components.
Or I missed something ?

flomillot commented on demo/src/app.js:
```diff
@@ -394,6 +412,13 @@ const AppContent = ({ language, onLanguageClick }) => {
     );
 
+    const CustomTreeViewFinder =
+        stylesProvider === 'emotion'
```
Why do we keep both way ?

jonenst replied:
I made a backwards compatible migration. It's still possible to generate classnames with jss, and then use the classes APIs to apply the classes. This is to make migrations easier. It's what mui did. So in the demo, we can test both modes. If mui removes the classes API, we will have to drop support for js, but for now it's easy to support

flomillot commented on demo/src/right-resizable-box.js:
```diff
@@ -38,19 +39,20 @@ const useStyles = makeStyles((theme) => ({
-const RightResizableBox = (props) => {
-    const classes = useStyles();
+// TODO can we avoid to define a component just to add sx support ?
+const ResizableBoxSx = styled(ResizableBox)({});
```
Instead of doing this and using mergeSx, the documentation suggest to do it in an other way, did you see it ?
https://mui.com/material-ui/customization/how-to-customize/#dynamic-overrides

sBouzols commented on package.json:
```diff
@@ -39,7 +39,6 @@
         "@mui/icons-material": "^5.5.0",
         "@mui/lab": "^5.0.0-alpha.72",
         "@mui/material": "^5.5.0",
-        "@mui/styles": "^5.5.0",
```
remove it from devDependencies as well ?

sBouzols replied:
I see you want to keep it for the demo app to compare ok.

sBouzols commented on demo/src/right-resizable-box.js:
```diff
@@ -66,27 +68,26 @@ const RightResizableBox = (props) => {
-            className={clsx(classes.panel, {
-                [classes.resizePanelHandle]: !props.disableResize,
-            })}
+            sx={mergeSx(
```
cf doc : https://www.npmjs.com/package/react-resizable?activeTab=readme
it says : "Unknown props that are not in the list below will be passed to the child component."
nor `className` nor `sx` are a known prop, maybe set it directly to the Box child ?
Or I miss something ?

jonenst replied:
sx is handled by styled and transformed to a classname prop on the component.
Not sure what's best between setting them on the resizable or the box

sBouzols commented on demo/src/app.js:
```diff
@@ -623,6 +657,28 @@ const AppContent = ({ language, onLanguageClick }) => {
+                                    labelId="styles-provider-label"
+                                    id="styles-provider"
+                                    value={stylesProvider}
+                                    label="Styles Provider"
```
missing translation

sBouzols commented on demo/src/app.js:
```diff
@@ -623,6 +657,28 @@ const AppContent = ({ language, onLanguageClick }) => {
+                                size="small"
+                            >
+                                <InputLabel id="styles-provider-label">
+                                    Custom theme
```
missing translation

EstherDarkish replied:
Added.

jonenst commented on src/components/OverflowableText/overflowable-text.js:
```diff
@@ -22,50 +21,57 @@ const overflowStyle = (theme) => ({
+};
 
-const useStyles = makeStyles(overflowStyle);
+export const OverflowableText = styled(
```
I used styled to add sx support on the component to provide an alternative to the className API

jonenst commented on src/components/MuiVirtualizedTable/ColumnHeader.js:
```diff
@@ -40,52 +41,50 @@ const useStyles = makeStyles((theme) => ({
     return (
-        <div className={clsx(classes.sortDiv)} onClick={props.onClick}>
+        <Box sx={styles.sortDiv} onClick={props.onClick}>
             {props.signedRank >= 0 ? (
-                <ArrowUpwardIcon className={clsx(visibilityClass)} />
+                <ArrowUpwardIcon sx={visibilityStyle} />
             ) : (
-                <ArrowDownwardIcon className={clsx(visibilityClass)} />
+                <ArrowDownwardIcon sx={visibilityStyle} />
             )}
             {sortRank > 1 && !props.hovered && <sub>{sortRank}</sub>}
-        </div>
+        </Box>
     );
 };
 
 const FilterButton = (props) => {
-    const classes = useStyles();
-    const visibilityClass =
+    const visibilityStyle =
         !props.filterLevel &&
-        (props.headerHovered ? classes.hovered : classes.transparent);
+        (props.headerHovered ? styles.hovered : styles.transparent);
```
didn't do it because it's not necessary in line 80 wheareas it's necessary at line 63... because line 80 uses mergeSx because we want to combine multiple rules, but line 63 only has the visibility rules

jonenst replied:
But it could be done
